### PR TITLE
feat: add date picker filter to statistics page

### PR DESCRIPTION
**Pull request by nikaaru**

### CODE REVIEW

**Comment by RanGojo** [comment]
Please change modal height from fixed height to flexible one to only wrap content without extra space.

**Comment by mikasackermn** [comment]
Screenshot from 2024-06-11 23-49-19
The chart's day display becomes jumbled when we filter by day.

File: components/common/PopoverDatePicker/index.tsx
Comment (mikasackermn) [inline components/common/PopoverDatePicker/index.tsx:1]: Screenshot from 2024-06-12 00-33-53
The border color used for today is incorrect.

Comment (mikasackermn) [inline components/common/PopoverDatePicker/index.tsx:1]: and also please check text colors

File: pages/statistics.tsx [hunk 1]
```diff
@@ -107,11 +175,71 @@ function Statistics(props: PropsType) {
...
+                        <ChevronDownIcon
+                          size="12px"
+                          className={`group-hover/box:text-hoverIcon text-primary-500 ${
+                            hasDateHoverStyle ? '!text-primary-500' : ''
+                          }`}
+                        />
+                      </div>
+                    </>
+                  ) : (
+                    <>
```
[the remark this answers is not in the document]
Comment (mikasackermn) [reply pages/statistics.tsx:219]: Dropdown is only incompatible with desktop mode.

File: components/common/RangeDatePicker/CustomDayContent.tsx [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+          : ''
+      } ${
+        hasCurrentCellBackgroundHover(date, false)
+          ? 'bg-overlay rounded-s-full'
+          : ''
+      } ${
+        range?.from &&
+        range.to &&
+        isAfter(date, range.from) &&
+        isBefore(date, range?.to)
```
Comment (mikasackermn) [inline components/common/RangeDatePicker/CustomDayContent.tsx:80]: I believe it is preferable to include conditions such as these in a variable with the appropriate name.

File: components/common/RangeDatePicker/CustomDayContent.tsx [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+        hasFromCellBackgroundHover(date, true)
+          ? 'bg-overlay  rounded-e-full'
+          : ''
+      } ${
+        hasFromCellBackgroundHover(date, false)
+          ? 'bg-overlay rounded-s-full'
+          : ''
+      } ${
+        range?.to && date.getTime() === range?.to?.getTime()
+          ? 'bg-overlay rounded-e-full'
```
Comment (mikasackermn) [inline components/common/RangeDatePicker/CustomDayContent.tsx:66]: Also here

File: components/common/RangeDatePicker/CustomDayContent.tsx [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+        isAfter(date, range.from) &&
+        isBefore(date, range?.to)
+          ? 'bg-overlay'
+          : ''
+      } ${hasRangeCellBackgroundHover(date) ? 'bg-overlay' : ''}`}>
+      <div
+        className={`w-full h-full flex items-center justify-center rounded-full hover:bg-hoverBackground hover:text-neutral-400 ${
+          date.getTime() === range?.from?.getTime() ||
+          date.getTime() === range?.to?.getTime()
+            ? '!bg-secondary-500 !text-baseForeground'
```
Comment (mikasackermn) [inline components/common/RangeDatePicker/CustomDayContent.tsx:88]: Also here

**Review by mikasackermn — COMMENTED** [review]
Thanks. I've put some comments.

File: pages/statistics.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (nikaaru) [reply pages/statistics.tsx:219]: image

This is how it is designed in Figma.
I did not see this design that you sent.

File: components/common/PopoverDatePicker/index.tsx [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+  const handleApply = () => {
+    onApply(dateRange);
+  };
+
+  return (
+    <div
+      style={{
+        boxShadow: '5px 5px 10px 0px rgba(0, 0, 0, 0.10)',
+      }}
+      className={`md:absolute md:right-0 md:top-[45px] bg-baseForeground md:z-10 md:rounded-soft ${
```
[the remark this answers is not in the document]
Comment (nikaaru) [reply components/common/PopoverDatePicker/index.tsx:31]: Fixed

File: components/common/PopoverDatePicker/index.tsx
Comment (nikaaru) [reply components/common/PopoverDatePicker/index.tsx:1]: Fixed

File: components/common/RangeDatePicker/CustomDayContent.tsx (hunk 2 above)
Comment (nikaaru) [reply components/common/RangeDatePicker/CustomDayContent.tsx:80]: Fixed

File: components/common/RangeDatePicker/CustomDayContent.tsx (hunk 3 above)
Comment (nikaaru) [reply components/common/RangeDatePicker/CustomDayContent.tsx:66]: Fixed

File: components/common/RangeDatePicker/CustomDayContent.tsx (hunk 4 above)
Comment (nikaaru) [reply components/common/RangeDatePicker/CustomDayContent.tsx:88]: I fixed the rest, but I think this one is clear

**Review by mikasackermn — APPROVED** [review]
LGTM

**Pull request merged**